column_buffer.go

```diff
@@ -969,7 +974,11 @@ func (col *int32ColumnBuffer) WriteInt32s(values []int32) (int, error) {
 
 func (col *int32ColumnBuffer) WriteValues(values []Value) (int, error) {
 	var model Value
-	col.writeValues(makeArrayValue(values, unsafe.Offsetof(model.u64)), columnLevels{})
+	if cpu.IsBigEndian {
+		col.writeValues(makeArrayValue(values, (unsafe.Offsetof(model.u64)+4)), columnLevels{})
```

Comment:
Also curious about why this offset is 4.

Reply:
On a big endian system, an int32 or uint32 value, which is in little endian format, is byte aligned to the 4th byte in a u64 (8 bytes) variable.. This is will become 4th byte when interpreted as a little endian byte format. So, in order to access an int32/uint32 value out of u64 variable, we have to have an offset of "4"...

Reply:
Makes total sense!

Let's move these to constants and add this explanation as documentation so we record this context for future maintainers 👍

Reply:
I have added this as a comment as part of the code in a file.